File: lib/core/utils/time_utils.dart
String formatArrivalTime(int addMinutes) {
  final t = DateTime.now().add(Duration(minutes: addMinutes));
  return '${t.hour.toString().padLeft(2, '0')}:${t.minute.toString().padLeft(2, '0')}';
}

String formatWon(int amount) {
  if (amount == 0) return 'Free';
  return '₩${amount.toString().replaceAllMapped(RegExp(r'(\d{1,3})(?=(\d{3})+(?!\d))'), (m) => '${m[1]},')}';
}

Map<String, String> getKMABaseDateTime() {
  const forecastHours = [2, 5, 8, 11, 14, 17, 20, 23];
  // Data available ~10 min after the base hour
  final ref = DateTime.now().subtract(const Duration(minutes: 10));

  int? base;
  var date = DateTime(ref.year, ref.month, ref.day);

  for (final h in forecastHours.reversed) {
    if (ref.hour >= h) { base = h; break; }
  }

  // Before 02:10 KST → use yesterday's 2300
  if (base == null) {
    date = date.subtract(const Duration(days: 1));
    base = 23;
  }

  final dateStr =
      '${date.year}${date.month.toString().padLeft(2, '0')}${date.day.toString().padLeft(2, '0')}';
  final timeStr = '${base.toString().padLeft(2, '0')}00';
  return {'date': dateStr, 'time': timeStr};
}

String savedAgo(DateTime savedAt) {
  final d = DateTime.now().difference(savedAt).inMinutes;
  if (d < 60)   return '${d}m ago';
  if (d < 1440) return '${d ~/ 60}h ago';
  return '${d ~/ 1440}d ago';
}

File: src/utils/timeUtils.ts
export function formatArrivalTime(addMinutes: number): string {
  const d = new Date(Date.now() + addMinutes * 60_000)
  const h = String(d.getHours()).padStart(2, '0')
  const m = String(d.getMinutes()).padStart(2, '0')
  return `${h}:${m}`
}

export function formatKoreanWon(amount: number): string {
  if (amount === 0) return 'Free'
  return `₩${amount.toLocaleString()}`
}

const FORECAST_HOURS = [2, 5, 8, 11, 14, 17, 20, 23]

export function getKMABaseDateTime(): { date: string; time: string } {
  const now = new Date()
  const hour = now.getHours()
  const minute = now.getMinutes()

  let baseHour = FORECAST_HOURS[0]
  for (const h of FORECAST_HOURS) {
    if (hour > h || (hour === h && minute >= 10)) {
      baseHour = h
    }
  }

  const pad = (n: number) => String(n).padStart(2, '0')
  const y = now.getFullYear()
  const mo = pad(now.getMonth() + 1)
  const d = pad(now.getDate())

  return {
    date: `${y}${mo}${d}`,
    time: `${pad(baseHour)}00`,
  }
}
